lib/src/solver/assignment.dart

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import '../package_name.dart';
+import 'incompatibility.dart';
+import 'term.dart';
+
+/// A term in a [PartialSolution] that tracks some additional metadata.
+class Assignment extends Term {
+  /// The number of decisions at or before this in the [PartialSolution] that
+  /// contains it.
+  final int decisionLevel;
+
+  /// The index of this assignment in [PartialSolution.assignments].
+  final int index;
+
+  /// The incompatibility that caused this assignment to be derived, or `null`
+  /// if it isn't a derivation.
+  final Incompatibility cause;
+
+  Assignment(
+      PackageName package, bool isPositive, this.decisionLevel, this.index,
```

Comment:
I *could*, but I'd rather not. This is a case where

1. There's no obvious default,
2. every caller is expected to pass in the value, and
3. the semantics of the boolean are clear (since the whole point of a term is associating a package range with a boolean).